service/src/main/java/com/epam/esm/service/impl/OrderServiceImpl.java

```diff
@@ -60,46 +52,99 @@ public Order create(Long userId, Long giftCertificateId) {
         return orderRepository.save(order);
     }
 
-    /**
-     * {@inheritDoc}
-     */
     @Override
     public List<Order> findAllWithPage(int page, int size) {
-        try {
-            return orderRepository.findAllWithPage(page, size);
-        } catch (ValidationException e) {
-            throw new ValidationException("Page number and page size must be positive");
+        validatePageAndSize(page, size);
+        return orderRepository.findAllWithPage(page, size);
+    }
+
+    @Override
+    public Order findById(Long id, Authentication authentication) {
+        Order order = orderRepository.findById(id)
```

Comment:
First of all, I would advise checking the user if he/she has access to get an order or not. Then you should check that this user admin or user id corresponds with the user id in Order.